core/store/models/common.go

```diff
@@ -425,20 +425,41 @@ func (c Cron) String() string {
 	return string(c)
 }
 
-// Duration is a time duration.
-type Duration time.Duration
+// Duration is a non-negative time duration.
+type Duration struct{ d time.Duration }
+
+func NewDuration(d time.Duration) (Duration, error) {
```

Comment:
`New` is typically used to indicate returning a pointer, `Make` is used to denote copies.